File: Java/Casing.java
import java.util.ArrayList;
import java.util.List;

// Kelas Casing dengan komposisi (memiliki Fan objects)
public class Casing extends Komponen {
    private String formFactor; // ATX, MicroATX, etc
    private String material;   // Tempered Glass, Metal, etc
    private List<Fan> fans;

    public Casing() {
        super();
        this.formFactor = "";
        this.material = "";
        this.fans = new ArrayList<>();
    }

    public Casing(String formFactor, String material, List<Fan> fans, String merk, String nama, int harga) {
        super(merk, nama, harga);
        this.formFactor = formFactor;
        this.material = material;
        this.fans = fans;
    }

    public void setFormFactor(String formFactor) {
        this.formFactor = formFactor;
    }

    public void setMaterial(String material) {
        this.material = material;
    }

    public void setFans(List<Fan> fans) {
        this.fans = fans;
    }

    public void addFan(Fan fan) {
        this.fans.add(fan);
    }

    public String getFormFactor() {
        return this.formFactor;
    }

    public String getMaterial() {
        return this.material;
    }

    public List<Fan> getFans() {
        return this.fans;
    }

    @Override
    public void displayInfo() {
        super.displayInfo();
        System.out.println("Form Factor: " + this.formFactor);
        System.out.println("Material: " + this.material);
        System.out.println("Jumlah Fan: " + this.fans.size());

        if (!this.fans.isEmpty()) {
            System.out.println("Detail Fan:");
            for (int i = 0; i < this.fans.size(); i++) {
                System.out.println("  Fan #" + (i + 1) + ":");
                this.fans.get(i).displayInfo();
                System.out.println();
            }
        }
    }
}

File: Python/Casing.py
from Komponen import Komponen
from Cooling import Fan

# Kelas Casing dengan komposisi (memiliki Fan objects)
class Casing(Komponen):
    def __init__(self, form_factor="", material="", fans=None, merk="", nama="", harga=0):
        super().__init__(merk, nama, harga)
        self.form_factor = form_factor
        self.material = material
        self.fans = fans if fans is not None else []
        
    def set_form_factor(self, form_factor):
        self.form_factor = form_factor
        
    def set_material(self, material):
        self.material = material
        
    def set_fans(self, fans):
        self.fans = fans
        
    def add_fan(self, fan):
        self.fans.append(fan)
        
    def get_form_factor(self):
        return self.form_factor
        
    def get_material(self):
        return self.material
        
    def get_fans(self):
        return self.fans
        
    def display_info(self):
        super().display_info()
        print(f"Form Factor: {self.form_factor}")
        print(f"Material: {self.material}")
        print(f"Jumlah Fan: {len(self.fans)}")
        
        if self.fans:
            print("Detail Fan:")
            for i, fan in enumerate(self.fans, 1):
                print(f"  Fan #{i}:")
                fan.display_info()
                print()
